lib/tasks/db/set_polyglot_language_weights.rake

```diff
@@ -15,30 +15,59 @@ namespace :db do
     ver1&.split('.')&.map(&:to_i) <=> ver2&.split('.')&.map(&:to_i)
   end
 
-  def language_compare(lang1, lang2)
+  @latest_hash = {}
+  def latest?(language)
+    unless @latest_hash.key?(language.name)
+      latest_language = language
+      @all_languages.each do |other_language|
+        next unless latest_language.polyglot_name == other_language.polyglot_name
+
+        @latest_hash[other_language.name] = false
+        if version_compare(latest_language.polyglot_version, other_language.polyglot_version) < 0
+          latest_language = other_language
+        end
+      end
+      @latest_hash[latest_language.name] = true
+    end
+
+    @latest_hash[language.name]
+  end
```

Comment:
overly complex with redundant looping over other languages.  
`Coursemology::Polyglot::Language.all.group_by(&:polyglot_name).each do |polyglot_name, languages|`

Something like this can help you group by language first. Then get the max of each group to and put into your hash. Anything language/key not in the hash will be treated as not latest.

Reply:
Reworked the `@latest_hash` table initialization.